pkg/types/types.go

```diff
@@ -234,40 +267,49 @@ func (s byStepTime) Less(i, j int) bool {
 	return s[i].StepTime < s[j].StepTime
 }
 
-func mergeMetrics(metrics []Metric) Metric {
+func mergeMetrics(metrics []Metric, mismatchCheck bool) (metric Metric, mismatches int) {
 	if len(metrics) == 0 {
-		return Metric{}
+		return Metric{}, 0
 	}
 
 	if len(metrics) == 1 {
-		return metrics[0]
+		m := metrics[0]
+		return m, 0
 	}
 
 	sort.Sort(byStepTime(metrics))
 	healed := 0
 
 	// metrics[0] has the highest resolution of metrics
-	metric := metrics[0]
+	metric = metrics[0]
 	for i := range metric.Values {
-		if !metric.IsAbsent[i] {
-			continue
-		}
-
-		// found a missing value, look for a replacement
+		pointExists := !metric.IsAbsent[i]
+		areValuesEqual := true
 		for j := 1; j < len(metrics); j++ {
+			if pointExists && !mismatchCheck {
+				break
+			}
 			m := metrics[j]
 
 			if m.StepTime != metric.StepTime || len(m.Values) != len(metric.Values) {
 				break
 			}
 
-			// found one
-			if !m.IsAbsent[i] {
+			if m.IsAbsent[i] {
+				continue
+			}
+
+			if !pointExists {
 				metric.IsAbsent[i] = m.IsAbsent[i]
 				metric.Values[i] = m.Values[i]
 				healed++
-				break
+				pointExists = true
 			}
+
+			areValuesEqual = areValuesEqual && metric.Values[i] == m.Values[i]
```

Comment:
If short-circuiting is added in the beginning of the loop, then `areValuesEqual` check could be dropped here